test/jdk/java/awt/Graphics2D/DrawString/IgnoredWhitespaceTest.java

```diff
@@ -0,0 +1,185 @@
+        g2d.setColor(Color.BLACK);
+        g2d.drawString(new AttributedString(text, Map.of(TextAttribute.FONT, font)).getIterator(), x, y);
+        actual = findTextBoundingBox(image);
+        assertEqual(expected, actual, text);
+
+        g2d.setColor(Color.WHITE);
+        g2d.fillRect(0, 0, w, h);
+        g2d.setColor(Color.BLACK);
+        g2d.drawChars(text.toCharArray(), 0, text.length(), x, y);
+        actual = findTextBoundingBox(image);
+        assertEqual(expected, actual, text);
+
+        g2d.setColor(Color.WHITE);
+        g2d.fillRect(0, 0, w, h);
+        g2d.setColor(Color.BLACK);
+        g2d.drawGlyphVector(font.createGlyphVector(frc, text), x, y);
+        actual = findTextBoundingBox(image);
+        assertEqual(expected, actual, text);
+    }
+
+    private static void assertEqual(Rectangle r1, Rectangle r2, String text) {
+        if (!r1.equals(r2)) {
+            String escaped = text.replace("\r", "\\r").replace("\n", "\\n").replace("\t", "\\t");
+            String msg = String.format("for text '%s': %s != %s", escaped, r1.toString(), r2.toString());
+            throw new RuntimeException(msg);
+        }
+    }
+
+    private static Font getPhysicalFont(int size) {
+        GraphicsEnvironment ge = GraphicsEnvironment.getLocalGraphicsEnvironment();
+        String[] names = ge.getAvailableFontFamilyNames();
+        for (String n : names) {
+            switch (n) {
+                case Font.DIALOG:
+                case Font.DIALOG_INPUT:
+                case Font.SERIF:
+                case Font.SANS_SERIF:
+                case Font.MONOSPACED:
+                     continue;
+                default:
+                    Font f = new Font(n, Font.PLAIN, size);
+                    if (f.canDisplayUpTo("AZaz09") == -1) {
+                        return f;
+                    }
+            }
+        }
+        return null;
+    }
+
+    private static Rectangle findTextBoundingBox(BufferedImage image) {
+        int minX = Integer.MAX_VALUE;
+        int minY = Integer.MAX_VALUE;
+        int maxX = Integer.MIN_VALUE;
+        int maxY = Integer.MIN_VALUE;
+        int width = image.getWidth();
+        int height = image.getHeight();
+        int[] rowPixels = new int[width];
+        for (int y = 0; y < height; y++) {
+            image.getRGB(0, y, width, 1, rowPixels, 0, width);
+            for (int x = 0; x < width; x++) {
+                boolean white = (rowPixels[x] == -1);
+                if (!white) {
+                    if (x < minX) {
+                        minX = x;
+                    }
+                    if (y < minY) {
+                        minY = y;
+                    }
+                    if (x > maxX) {
+                        maxX = x;
+                    }
+                    if (y > maxY) {
+                        maxY = y;
+                    }
+                }
+            }
+        }
+        if (minX != Integer.MAX_VALUE) {
```

Comment:
```suggestion
        }

        if (minX != Integer.MAX_VALUE) {
```
A blank line here breaks a long method into sections.